src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PopupWindow.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using UnityEditor;
+using UnityEngine;
+
+namespace GitHub.Unity
+{
+    [Serializable]
+    class PopupWindow : BaseWindow
+    {
+        public enum PopupViewType
+        {
+            PublishView,
+            AuthenticationView
+        }
+
+        [NonSerialized] private Subview activeView;
+
+        [SerializeField] private PopupViewType activeViewType;
+        [SerializeField] private AuthenticationView authenticationView;
+        [SerializeField] private PublishView publishView;
+
+        [MenuItem("GitHub/Authenticate")]
+        public static void Launch()
+        {
+            var popupWindow = Open(PopupViewType.AuthenticationView);
+            popupWindow.InitializeWindow(EntryPoint.ApplicationManager);
+        }
+
+        public static PopupWindow Open(PopupViewType popupViewType, Action<bool> onClose = null)
+        {
+            var popupWindow = GetWindow<PopupWindow>(true);
+            if (onClose != null)
+            {
```

Comment:
So the thing with the `GetWindow` method is that it'll return an existing window if there's one already. That means that we need to decide whether we need to reset the window and how. If there's another thing waiting for the close callback. we need to make sure it gets closure on it, probably by having the window shut itself down with onClose(false) and then resetting all the data and state.

This `Open` method should probably do all the work, including calling `InitializeWindow` (which is currently being called in `Launch` above. The PopupWindow class could probably use a Reset method to cancel any callbacks and clear all data before anything else, which it can probably do in it's `InitializeWindow` implementation.

`InitializeWindow` should probably also be called before calling show on the window, right now it's the other way around.